clang/lib/CIR/CodeGen/CIRGenBuiltin.cpp

```diff
@@ -1649,8 +1649,13 @@ RValue CIRGenFunction::buildBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
 
   case Builtin::BI__builtin_signbit:
   case Builtin::BI__builtin_signbitf:
-  case Builtin::BI__builtin_signbitl:
-    llvm_unreachable("BI__builtin_signbit like NYI");
+  case Builtin::BI__builtin_signbitl: {
+    mlir::Location Loc = getLoc(E->getBeginLoc());
+    mlir::Value val = buildScalarExpr(E->getArg(0));
+    auto resTy = mlir::cir::IntType::get(builder.getContext(), 32, true);
+    auto ret = builder.create<mlir::cir::SignBitOp>(Loc, resTy, val);
```

Comment:
Since the result type is always `int32`, I recommend adding a new builder for `SignBitOp` in `CIRBaseBuilder` and create the result type there.